omf-bifrost/src/admin/config.rs

```diff
@@ -0,0 +1,531 @@
+use anyhow::{Context, Result};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::fs;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+#[serde(rename_all = "snake_case")]
+pub enum HighwayType {
+    Trunk,
+    TrunkLink,
+    Track,
+    Footway,
+    Pedestrian,
+    Bridleway,
+    Cycleway,
+    Path,
+    Motorroad,
+}
+
+#[repr(u8)]
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq, Hash)]
+#[serde(rename_all = "snake_case")]
+pub enum AccessMode {
+    Auto = 0,
+    Pedestrian = 1,
+    Bicycle = 2,
+    Truck = 3,
+    Emergency = 4,
+    Taxi = 5,
+    Bus = 6,
+    Hov = 7,
+    Wheelchair = 8,
+    Moped = 9,
+    Motorcycle = 10,
+}
+
+impl AccessMode {
+    pub fn bit(self) -> i64 {
+        1 << (self as u8)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct AdminConfig {
+    pub allow_intersection_names: HashMap<String, bool>,
+    pub admin_access: HashMap<String, HashMap<HighwayType, Vec<AccessMode>>>,
+}
+
+impl Default for AdminConfig {
+    fn default() -> Self {
+        use self::{AccessMode as M, HighwayType as H};
+
+        let allow_intersection_names = HashMap::from([
+            ("JP".to_string(), true),
+            ("KP".to_string(), true),
+            ("KR".to_string(), true),
+            ("NI".to_string(), true),
+        ]);
+
+        let admin_access = HashMap::from([
```

Comment:
Was this sourced from a Valhalla default somewhere? If so, it might be good to link to the upstream